src/config.c

```diff
@@ -3068,7 +3068,8 @@ standardConfig static_configs[] = {
     createBoolConfig("aof-disable-auto-gc", NULL, MODIFIABLE_CONFIG | HIDDEN_CONFIG, server.aof_disable_auto_gc, 0, NULL, updateAofAutoGCEnabled),
     createBoolConfig("replica-ignore-disk-write-errors", NULL, MODIFIABLE_CONFIG, server.repl_ignore_disk_write_error, 0, NULL, NULL),
     createBoolConfig("extended-redis-compatibility", NULL, MODIFIABLE_CONFIG, server.extended_redis_compat, 0, NULL, updateExtendedRedisCompat),
-    createBoolConfig("enable-debug-assert", NULL, IMMUTABLE_CONFIG | HIDDEN_CONFIG, server.enable_debug_assert, 0, NULL, NULL),
+    createBoolConfig("debug-assert-enabled", NULL, IMMUTABLE_CONFIG | HIDDEN_CONFIG, server.debug_assert_enabled, 0, NULL, NULL),
+    createBoolConfig("io_uring-enabled", NULL, DEBUG_CONFIG | MODIFIABLE_CONFIG, server.io_uring_enabled, 0, NULL, NULL),
```

Comment:
Can we detect if io_uring is supported on the system and add it as a validation step?

Reply:
Any benefit to using validation function vs apply function?
 
We decided to align it with how TLS is handled. If it's not installed in the system, log and exit. https://github.com/valkey-io/valkey/pull/599#discussion_r1639823084

Let's do this here too, similar to how it's done for TLS in the function `applyTlsCfg`.

Reply:
OTOH, this is a modifiable config, so I think the validation is a better idea. If someone tries to enable it with CONFIG SET, it is rejected if it's not supported on the system.

Reply:
hmm - any benefit of allowing this to be modifiable? My mental model is io-threads.

Reply:
@zuiderkwast @madolson @PingXie To keep it simple, I changed the `io-uring-enabled` to IMMUTABLE_CONFIG. the detection will be done through `io_uring_queue_init_params` when server starts.